test/Gen.hs

```diff
@@ -220,23 +231,29 @@ showing = Pure . flip showsPrec <*> id
 showingFn :: (Show a, Show b) => Fn.Fn a b -> Term (a -> b)
 showingFn = Pure . flip showsPrec <*> Fn.apply
 
-atom :: String -> a -> Gen a
-atom s = Gen . pure . Pure (const (showString s))
 
-label :: String -> a -> Gen a
+type GenTerm = WriterT (Set.Set LabelName) Gen :.: Term
```

Comment:
This turns out to do everything our hand-rolled `Gen` did, while also allowing us to easily wrap up `subtermM`/`subtermM2` 🎉